src/React.Core/Babel.cs

```diff
@@ -350,12 +350,36 @@ string filename
 		)
 		{
 			var outputPath = GetOutputPath(filename);
-			var sourceMapPath = GetSourceMapOutputPath(filename);
 			var contents = _fileSystem.ReadAsString(filename);
-			var result = TransformWithHeader(filename, contents, null);
-			_fileSystem.WriteAsString(outputPath, result.Code);
-			_fileSystem.WriteAsString(sourceMapPath, result.SourceMap == null ? string.Empty : result.SourceMap.ToJson());
-			return outputPath;
+            if (MustTranspileFile(contents, outputPath))
+            {
+                var result = TransformWithHeader(filename, contents, null);
+
+                var sourceMapPath = GetSourceMapOutputPath(filename);
+                _fileSystem.WriteAsString(outputPath, result.Code);
+                _fileSystem.WriteAsString(sourceMapPath, result.SourceMap == null ? string.Empty : result.SourceMap.ToJson());                
+            }
+		    return outputPath;
 		}
+
+        /// <summary>
+        /// Checks whether an input file (given as inputFileContents) should be transpiled
+        /// by calculating the hash and comparing it to the hash value stored
+        /// in the file given by outputPath. If the outputPath file does not
+        /// exist the input file should always be transpiled.
+        /// </summary>
+        /// <param name="inputFileContents">The contents of the input file.</param>
+        /// <param name="outputPath">The output path of the (possibly previously) generated file.</param>
+        /// <returns>Returns true if the file should be transpiled, false otherwise.</returns>
+	    public virtual bool MustTranspileFile(string inputFileContents, string outputPath)
```

Comment:
ValidateCache is rather vague and unspecific - CacheIsValid would be a better name, since it better hints at what state the cache is actually in, when true/false is returned.

And I like flipping the true/false values, since it would enable me to turn around the if in your next comment ;)